code/game/objects/items/quickdeploy_cade.dm

```diff
@@ -0,0 +1,37 @@
+//An item that allows a very quick deployment of a barricade
+/obj/item/quikdeploy_cade
+	name = "QuikDeploy Barricade"
+	desc = "This is a QuikDeploy barricade, allows for extremely fast laying of barricades. Usually found in specific vendors and cannot be picked back up."
+	icon = 'icons/obj/items/quikdeploy_cade.dmi'
+	icon_state = "metal"
+	var/obj/thing_to_deploy = /obj/structure/barricade/metal
+	w_class = WEIGHT_CLASS_SMALL //While this is small, normal 50 stacks of metal is NORMAL so this is a bit on the bad space to cade ratio
+
+/obj/item/quikdeploy_cade/examine(mob/user)
+	..()
+	to_chat(user, "This QuikDeploy cade seems to deploy a [thing_to_deploy.name].")
+
+/obj/item/quikdeploy_cade/attack_self(mob/user)
+	to_chat(user, "You start to deploy the cade on your tile in front of you...")
+	if(!do_after(usr, 2 SECONDS, TRUE, src, BUSY_ICON_BUILD))
+		to_chat(user, "You decide against putting the barricade here.")
+		return
+	var/check_build = TRUE
+	for(var/obj/thing in get_turf(src))
+		if(!istype(thing, thing_to_deploy))
```

Comment:
It should check for dense turfs, dense non-on_border objects, and dense on_border objects with the same dir. Else you can stack different types of barricades and windows.

Reply:
Addressed that

Reply:
Here's an example on how to do it: https://github.com/tgstation/TerraGov-Marine-Corps/pull/3566/files#diff-5b80535f483d4c008a0b2c4532a29f15R51